Classes/CocoaLumberjack.swift

```diff
@@ -71,38 +69,100 @@ public func resetDefaultDebugLevel() {
     resetDynamicLogLevel()
 }
 
+/// If `true`, all logs (except errors) are logged asynchronously by default.
+public var asyncLoggingEnabled = true
```

Comment:
Too ObjC-ish. Not enough nested classes/generics for Swiftish way :)

However, are you sure that this variable should be so alone?
I suppose that access should be defined by the structure or by the class level.
`AsyncLoggingSettings.Enabled` feels better. And word `Settings` points to something that I am able to change.
However, it doesn't sync access on read/write. Is it ok?

Also, here we can define more complex Swift-ish tricks and whistles - we can define a class with static functions for logging.
And later, we can define a global wrappers that call these functions from this class.

```swift
class DDLogSwiftish {
   // Define settings here.
   class Settings {
       static public var asyncLoggingEnabled = true
    }
   // And functions
   @inlineable public static verbose(params: [Int]) {}
   // others.
}

@inlinable
public func DDLogVerbose(params: [Int]) {
    DDLogSwiftish.verbose(params: params)
}
```

Reply:
Simple question: why introduce a simple new feature so complicated?

`dynamicLogLevel` (formerly known as `defaultDebugLevel`) has been defined like this forever. If you want to know where it belongs to, simply assign it as `CocoaLumberjackSwift.asyncLoggingEnabled = true`. Swift does allow that.

What you describe is a complete refactoring of the Swift interface of CocoaLumberjack. There are a few nice ideas for that, but it's not the goal of this PR. And if we did so, I'd get rid of the global `func`s and DD-prefixes completely and use something like `Log.error(...)`, `Log.verbose(...)`. But this would also be out of scope for 3.5.0 which should be released ASAP.

Reply:
@ffried 
Simple answer - simple decisions today _will_ affect problems tomorrow.
You add one variable that _is_ neither have sync access nor have proper scope ( global scope ). Is it really _good_ practice? I can't remember Cocoas framework which uses it. Why mature opensource use it?

Again, when you define this function, person will `import CocoaLumberjack` as always and this person expects that settings ( various settings, even one setting ) are consolidated in one `facade` ( ah, patterns ) that has name `DDLogSettings`.

Reply:
Don't get me wrong. I agree that currently the Swift interface does look a lot like ObjC. I just don't see why I should refactor the whole interface just to implement one small feature.
It's not good Swift practice, no. But it's not always clever to throw everything away just to do something small. If you feel better, I'll open another PR with refactoring of CocoaLumberjackSwift. But I won't stuff it into this one.

But since will be a breaking change (as opposed to this change), it'll have to target a 4.0.0 release.

Reply:
@ffried 
Well, I suggest to update release version of this feature to 4.0.0.

OR

I suggest to rename this variable to `DDLogAsyncLoggingEnabled` (`DDLogSettingsAsyncLoggingEnabled`) to emphasize further refactoring.

Reply:
The feature is simple enough and with our release plans, 4.0.0 is IMHO too far away for this feature to wait.

Renaming is IMHO also not the best idea, since it would also mean that `dynamicLogLevel` would need to be renamed and `currentFileName(_:)` probably too. I implemented it so that it matches the current state of the Swift interface. After the refactoring, Swift will tell anyone using the "old" interface, because I'll add the correct deprecation attributes.

Reply:
@ffried 
Hm...
``` dynamicLogLevel ``` for each `import CocoaLumberjack`
Hm...

Ok, if it necessary, so, do it.